types/sharded_map.go

```diff
@@ -0,0 +1,110 @@
+/*
+ * SPDX-FileCopyrightText: © Hypermode Inc. <hello@hypermode.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// Package types contains some very common utilities used by Dgraph. These utilities
+// are of "miscellaneous" nature, e.g., error checking.
+package types
+
+import (
+	"sync"
+)
+
+type ShardedMap struct {
+	Shards []map[uint64]Val
```

Comment:
can we not export it, it seems that we are only using it for finding the number of shards?